### PR TITLE
Use current directory if targeted repo

### LINKED ISSUE #42: Support running upgrade provider from an already cloned repository
Enable running upgrade-provider against a repository already cloned, e.g.: locally or via `@actions/checkout`.

### PR DESCRIPTION
**Pull request by aq17**
Fixes https://github.com/pulumi/upgrade-provider/issues/42
If we run `upgrade-provider pulumi-{provider}` from within `pulumi-{provider}`, we should default to using the current repo instead of cloning

### CODE REVIEW

File: upgrade/steps.go [hunk 1]
```diff
@@ -310,7 +310,6 @@ func InformGitHub(
 		panic("Unknown action")
 	}
 	createPR := step.Cmd(exec.CommandContext(ctx, "gh", "pr", "create",
-		"--assignee", "@me",
```
Comment (aq17) [inline upgrade/steps.go:301]: For CI, errors on `could not find user pulumi-bot`. Removing for now to test

File: upgrade/steps.go [hunk 2]
```diff
@@ -298,7 +308,7 @@ func InformGitHub(
...
+		"--add-assignee", "@me",
```
Comment (iwahbe) [inline upgrade/steps.go:311]: Why this change?

File: upgrade/steps.go [hunk 3]
```diff
@@ -140,7 +140,17 @@ func ensureUpstreamRepo(ctx Context, repoPath string) step.Step {
...
+			if err != nil {
+				return "", fmt.Errorf("could not resolve current directory: '%s", err)
+			}
+			if idx := strings.Index(cwd, filepath.Base(repoPath)); idx != -1 {
+				idx = idx + len(expectedLocation) - 1
+				expectedLocation = cwd[:idx]
+			} else {
+				expectedLocation = filepath.Join(ctx.GoPath, "src", expectedLocation)
+
+			}
```
Comment (iwahbe) [inline upgrade/steps.go:153]: Can you explain what this is trying to do. It seems like the logic should be
```go
if filepath.Base(repoPath) == filepath.Base(cwd) && filepath.Base(cwd) != "" {
    	expectedLocation = cwd
else {
    	expectedLocation = filepath.Join(ctx.GoPath, "src", expectedLocation)
}
```

Comment (aq17) [reply upgrade/steps.go:153]: I think I was trying to account for the possibility of being in a subdirectory, i.e. `user/home/pulumi/pulumi-docker/provider`would set the expected location to `user/home/pulumi/pulumi-docker`

File: upgrade/steps.go (hunk 2 above)
Comment (aq17) [reply upgrade/steps.go:311]: Accident :(

File: upgrade/steps-helpers_test.go [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetRepoExpectedLocation(t *testing.T) {
+	ctx := Context{
+		GoPath: "Users/myuser/go",
+	}
+
+	cwd, err := os.Getwd()
+	assert.Nil(t, err)
```
Comment (iwahbe) [inline upgrade/steps-helpers_test.go:18]: Is there a way we can do this so that our tests don't depend on the current working directory? We could make `getRepoExpectedLocation` a pure function: `func(goPath, cwd, repoPath string) string`.

File: upgrade/steps-helpers_test.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+	mockRepoPath = trimSeparators(mockRepoPath)
+	expected = trimSeparators(expected)
+	assert.Equal(t, mockRepoPath, expected)
+
+	// test cwd completely different from repo path
+	mockRepoPath = filepath.Join("github.com", "pulumi", "random-provider")
+	expected, err = getRepoExpectedLocation(ctx, mockRepoPath)
+	expected = trimSeparators(expected)
+	assert.Nil(t, err)
+	assert.Equal(t, filepath.Join(ctx.GoPath, "src", mockRepoPath), expected)
```
Comment (iwahbe) [inline upgrade/steps-helpers_test.go:41]: This pattern is perfect for table based testing. That would look something like this:
```suggestion
	tests := []struct{ repoPath, expected string }{
		{ cwd, trimSeparators(expected) },
		{ strings.TrimSuffix(cwd, filepath.Base(cwd)), getRepoExpectedLocation(ctx, mockRepoPath) },
		{ filepath.Join("github.com", "pulumi", "random-provider"), trimSeparators(expected) },
	}
	
	for _, tt := range tests {
		tt := tt
		t.Run(fmt.Sprintf("(%s,%s)",tt.repoPath,tt.expected), func(t *testing.T) {
			mockRepoPath = filepath.Join("github.com", "pulumi", "random-provider")
			expected, err = getRepoExpectedLocation(ctx, mockRepoPath)
			expected = trimSeparators(expected)
			assert.Nil(t, err)
			assert.Equal(t, tt.expected, expected)
		})
	// test cwd completely different from repo path

```

Comment (aq17) [reply upgrade/steps-helpers_test.go]: Thanks for the suggestion, applied

**Review by iwahbe — APPROVED** [review]
Thanks for the work. I think this is much cleaner. LGTM!
